src/pipecat/services/ai_services.py

```diff
@@ -137,6 +141,13 @@ async def process_generator(self, generator: AsyncGenerator[Frame | None, None])
                     await self.push_frame(f)
 
 
+@dataclass
+class FunctionCallEntry:
+    function_name: Optional[str]
+    callback: Any  # TODO(aleix): add proper typing.
```

Comment:
just leaving a note here since this is marked as todo

Reply:
Currently the callbacks signatures are too long. I'm planning to create a data class and pass it as a single argument.